### PR TITLE
Handle custom attributes with empty values, fixes #270

### LINKED ISSUE #270: Error with customAttrCollapse:true and empty custom attributes
html-minifier fails, if you try to minify that:

```
<div ng-some
   style=""
></div>
```

The line break after the custom attribute is the problem.

Output:

```
TypeError: Cannot call method 'replace' of undefined
    at cleanAttributeValue (/Users/dsaenger/code/firmware/titan-web/titan-gui-web/node_modules/partial2js/node_modules/html-minifier/dist/htmlminifier.js:803:29)
    at normalizeAttribute (/Users/dsaenger/code/firmware/titan-web/titan-gui-web/node_modules/partial2js/node_modules/html-minifier/dist/htmlminifier.js:927:17)
    at Object.HTMLParser.start (/Users/dsaenger/code/firmware/titan-web/titan-gui-web/node_modules/partial2js/node_modules/html-minifier/dist/htmlminifier.js:1135:19)
    at parseStartTag (/Users/dsaenger/code/firmware/titan-web/titan-gui-web/node_modules/partial2js/node_modules/html-minifier/dist/htmlminifier.js:372:17)
    at String.replace (native)
    at new global.HTMLParser (/Users/dsaenger/code/firmware/titan-web/titan-gui-web/node_modules/partial2js/node_modules/html-minifier/dist/htmlminifier.js:236:22)
    at minify (/Users/dsaenger/code/firmware/titan-web/titan-gui-web/node_modules/partial2js/node_modules/html-minifier/dist/htmlminifier.js:1097:5)
```

### PR DESCRIPTION
**Pull request by ds82**
Fixes Bug #270 

Also added a test that triggers the described error without that fix.

### CODE REVIEW

File: src/htmlminifier.js [hunk 1]
```diff
@@ -250,7 +250,9 @@
       attrValue = attrValue.replace(/1\.0/g, '1').replace(/\s+/g, '');
     }
     else if (options.customAttrCollapse && options.customAttrCollapse.test(attrName)) {
-      attrValue = attrValue.replace(/\n+/g, '');
+      if ( !!attrValue ) {
```
Comment (duncanbeevers) [inline src/htmlminifier.js]: Do you mind moving this comparison up to the preceding else-if clause?

``` javascript
else if (attrValue && options.customAttrCollapse && options.customAttrCollapse.test(attrName)) {
```

**Comment by ds82** [comment]
@duncanbeevers: Of cause, you are right.
I updated the PR accordingly and rebased against current HEAD

**Comment by duncanbeevers** [comment]
@ds82 Looks like just a style change is necessary here to appease the jscs check.

**Comment by duncanbeevers** [comment]
:+1:

**Pull request merged**